plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/VacuumProcedure.java

```diff
@@ -137,6 +138,9 @@ public void vacuum(
         catch (TrinoException e) {
             throw e;
         }
+        catch (IOException e) {
+            throw new TrinoException(DELTA_LAKE_FILESYSTEM_ERROR, format("Failure when vacuuming %s.%s with retention %s: %s", schema, table, retention, e), e);
```

Comment:
This approach differs from a comment in the following `Exception` catch block. I think we should find a different solution.

Reply:
Any suggestion? We'd like to wrap these "known" errors as Trino `EXTERNAL_ERROR` for accounting purposes. This branch does that while leaving other unforeseen exceptions untouched, which will eventually be wrapped in a Trino `INTERNAL_ERROR`.

Reply:
This seems reasonable to me, and matches what we do in other places. File system errors should be categorized as `EXTERNAL_ERROR`.

@ebyhr did you have something different in mind?